protoc-gen-gogo/generator/generator.go

```diff
@@ -784,6 +784,18 @@ func (g *Generator) WrapTypes() {
 		genFileNames[n] = true
 	}
 	for _, f := range g.Request.ProtoFile {
+		// Make sure the registration path matches the import path.
+		// We check this by making sure that the file's name name
+		// starts with the package name.
+		//
+		// Example:
+		// Proto file "google/protobuf/descriptor.proto" should be imported
+		// from OS path ./google/protobuf/descriptor.proto.
+		expectedPrefix := strings.ReplaceAll(*f.Package, ".", "/") + "/"
+		if !strings.HasPrefix(*f.Name, expectedPrefix) {
+			panic(fmt.Errorf("file name %s does not start with expected %s; please make sure your folder structure matches the proto files fully-qualified names", *f.Name, expectedPrefix))
+		}
```

Comment:
Does this seems like a good enough check to you?

Reply:
I think so. @testinginprod what do you think?

Reply:
looks correct to me. Should we test it on the sdk?

Reply:
I did a branch on the SDK where I pointed to this PR, and it worked